tests/integration/security/reachability/context.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reachability
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/tests/integration/security/util"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+type TestCase struct {
+	ConfigFile string
+	Namespace  namespace.Instance
+
+	RequiredEnvironment environment.Name
+
+	// Indicates whether a test should be created for the given configuration.
+	Include func(src echo.Instance, opts echo.CallOptions) bool
+
+	// Handler called when the given test is being run.
+	OnRun func(ctx framework.TestContext, src echo.Instance, opts echo.CallOptions)
+
+	// Indicates whether the test should expect a successful response.
+	ExpectSuccess func(src echo.Instance, opts echo.CallOptions) bool
+}
+
+// Context is a context for reachability tests.
+type Context struct {
+	ctx                   framework.TestContext
+	g                     galley.Instance
+	p                     pilot.Instance
+	Namespace             namespace.Instance
+	A, B, Headless, Naked echo.Instance
+}
+
+// CreateReachabilityContext creates and initializes reachability context.
+func CreateContext(ctx framework.TestContext, g galley.Instance, p pilot.Instance) Context {
```

Comment:
No need to be public method I assume?

Reply:
caller in the same package.

Reply:
Keeping this public as the tests are now in different package. 

FYI, the plan is to have another sub folder for auto-mtls when the time come.